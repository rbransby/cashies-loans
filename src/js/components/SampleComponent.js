import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { assign } from 'lodash';

import * as SampleActions from '../actions/SampleActions';

class SampleComponent extends Component {
  constructor(props) {
    super(props);

    this.state = {
      logged: false
    };
  }

  signIn() {
    const { actions } = this.props;

    actions.signIn();
  }

  signOut() {
    const { actions } = this.props;

    actions.signOut();
  }

  render() {
    const { login } = this.props;

    return (
      <div>
        <h2>You are {`${login.logged ? 'logged in' : 'logged out'}`}</h2>
        {
          login.logged ?
          <button className="button button-secondary" onClick={this.signOut.bind(this)}>Sign out</button>
          :
          <button className="button" onClick={this.signIn.bind(this)}>Sign in</button>
        }
      </div>
    );
  }
}

function mapStateToProps(state/*, props*/) {
  return {
    login: state.SampleReducer.login
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(assign({}, SampleActions), dispatch)
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(SampleComponent);
