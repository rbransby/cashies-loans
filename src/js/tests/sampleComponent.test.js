import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Store from '../store';

import SampleComponent from '../components/sampleComponent';

const StoreInstance = Store();

test('Test the sample component', () => {
  const component = renderer.create(
    <Provider store={StoreInstance}>
      <SampleComponent />
    </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the function
  // Sign in
  tree.children[1].props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // Sign out
  tree.children[1].props.onClick();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

