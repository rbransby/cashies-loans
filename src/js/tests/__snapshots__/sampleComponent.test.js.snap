// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the sample component 1`] = `
<div>
  <h2>
    You are 
    logged out
  </h2>
  <button
    className="button"
    onClick={[Function]}
  >
    Sign in
  </button>
</div>
`;

exports[`Test the sample component 2`] = `
<div>
  <h2>
    You are 
    logged in
  </h2>
  <button
    className="button button-secondary"
    onClick={[Function]}
  >
    Sign out
  </button>
</div>
`;

exports[`Test the sample component 3`] = `
<div>
  <h2>
    You are 
    logged out
  </h2>
  <button
    className="button"
    onClick={[Function]}
  >
    Sign in
  </button>
</div>
`;
